storey/utils.py

```diff
@@ -42,6 +43,14 @@ def get_one_unit_of_duration(string_time):
     return multiplier
 
 
+def get_window_start_time_by_time(timestamp, window_millis):
+    midnight = datetime.utcfromtimestamp(int(timestamp/1000)).replace(hour=0, minute=0, second=0, tzinfo=timezone.utc)
```

Comment:
I know I've been harping on this one, but I'd feel better with just simple arithmetic, as is the case in the rest of the aggregation logic.
```
_seconds_per_day = 24 * 60 * 60


def get_window_start_time_by_time(timestamp, window_millis):
    timestamp_seconds = int(timestamp / 1000)
    midnight = int(timestamp_seconds / _seconds_per_day) * _seconds_per_day
    window_seconds = int(window_millis / 1000)
    return max(int(timestamp_seconds / window_seconds) * window_seconds, midnight) * 1000
```

Reply:
Also trying to understand why we hardcode midnight as opposed to just going by window (new requirement?).